effectful/ops/syntax.py

```diff
@@ -180,10 +181,17 @@ def _(t: Callable[P, T], *, name: Optional[str] = None) -> Operation[P, T]:
 
 @defop.register(Operation)
 def _(t: Operation[P, T], *, name: Optional[str] = None) -> Operation[P, T]:
+
+    @functools.wraps(t)
     def func(*args, **kwargs):
         raise NotImplementedError
 
-    functools.update_wrapper(func, t)
+    if name is None:
+        name = (
+            getattr(t, "__name__", str(t))[:10000]
+            + "__"
+            + uuid.uuid4().hex.split("-")[0]
```

Comment:
I don't think uuid does much for us here over using random.randint

Reply:
OK, I switched it to `randint`.